src/Export/CsvTarget.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @file
+ * Contains CSV implementation of ExtractionTargetInterface.
+ */
+
+namespace App\Export;
+
+use App\Document\Entry;
+use App\Document\ExtractionResult;
+use Box\Spout\Writer\Common\Creator\WriterEntityFactory;
+
+/**
+ * Class CsvTarget.
+ */
+class CsvTarget implements ExtractionTargetInterface
+{
+    /* @var \Box\Spout\Writer\WriterInterface $writer */
+    private $writer;
+    /* @var string $filename */
+    private $filename;
+    /* @var array $types */
+    private $types = null;
+
+    /**
+     * CsvTarget constructor.
+     *
+     * @param string $filename
+     */
+    public function __construct(string $filename)
+    {
+        $this->filename = $filename;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function initialize(): void
+    {
+        $this->writer = WriterEntityFactory::createCSVWriter();
+
+        $this->writer->openToFile($this->filename);
+
+        // Add headers.
+        $headers = [
+            'elasticId',
+            'date',
+            'clientId',
+            'agency',
+            'event',
+            'identifierType',
+            'materialId',
+            'response',
+            'imageId',
+        ];
+        $rowFromValues = WriterEntityFactory::createRowFromArray($headers);
+        $this->writer->addRow($rowFromValues);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function finish(): void
+    {
+        $this->writer->close();
```

Comment:
As an extra level you could also make a `__destruct()` function that calls this function to ensure proper file closer on errors/exceptions?

Reply:
added